src/python_be.cc

```diff
@@ -1818,10 +1898,13 @@ TRITONBACKEND_ModelInstanceExecute(
       LOG_MESSAGE(
           TRITONSERVER_LOG_ERROR,
           "Stub process is unhealthy and it will be restarted.");
+      instance_state->TerminateLogMonitor();
       instance_state->Stub()->KillStubProcess();
-      LOG_IF_ERROR(
-          instance_state->Stub()->Launch(),
-          "Failed to restart the stub process.");
+      TRITONSERVER_Error* err = instance_state->Stub()->Launch();
+      if(err == nullptr) {
+        instance_state->StartLogMonitor();
```

Comment:
Can this be part of `instance_state->Stub()->Launch()`?

Reply:
Will this be confusing later on that we are terminating/restarting the log thread in one file but starting it in another?

Reply:
Maybe we can combine both `Terminate` and `Start` in the in `instance_state->Stub()->Launch()`? Whenever `Stub()->Lanuch()` is called we will terminate the log monitor first and then Start it.

Reply:
Discussed offline. Resolving.